spec/openhab/dsl/rules/builder_spec.rb

```diff
@@ -1366,6 +1366,96 @@ def self.test_it(range, expected)
         test_it(10.minutes.ago..5.minutes.ago, false)
         test_it(1.day.ago.to_local_date..1.day.from_now.to_local_date, true)
       end
+
+      describe "Debounce Guards" do
+        before { Timecop.freeze(Time.at(0)) }
+
+        let(:item) { items.build { number_item "Item1" } }
+
+        def test_debouncing(type, arg, triggers, expected)
+          executed = []
+          rule do
+            updated Item1
+            send(type, arg)
+            run { |event| executed << event.state.to_i }
+          end
+
+          last_trigger = nil
+          last_executed = nil
+          triggers.chars.each_with_index do |trigger, i|
```

Comment:
:sobeautiful:

I saw you say you just copied the diagrams to the specs, but you're _actually_ generating the specs from said diagram. brilliant.